text/0000-pack-subcommands.md

```diff
@@ -0,0 +1,206 @@
+# Meta
+[meta]: #meta
+- Name: `pack` subcommands
+- Start Date: 2020-07-08
+- Author(s): @jromero
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: "N/A"
+
+# Summary
+[summary]: #summary
+
+Update the `pack` CLI to use more common subcommand patterns.
+
+# Motivation
+[motivation]: #motivation
+
+As the number of subcommands for `pack` grow, it's becoming more difficult to identify interworking subcommands at first glance. See the current list of subcommands below.
+
+```
+pack
+
+  build                 Generate app image from source code
+  rebase                Rebase app image with latest run image
+  inspect-image         Show information about a built image
+  set-run-image-mirrors Set mirrors to other repositories for a given run image
+  set-default-builder   Set default builder used by other commands
+  inspect-builder       Show information about a builder
+  suggest-builders      Display list of recommended builders
+  trust-builder         Trust builder
+  untrust-builder       Stop trusting builder
+  list-trusted-builders List Trusted Builders
+  create-builder        Create builder image
+  package-buildpack     Package buildpack in OCI format.
+  suggest-stacks        Display list of recommended stacks
+  version               Show current 'pack' version
+  report                Display useful information for reporting an issue
+  completion            Outputs completion script location
+  help                  Help about any command
+```
+
+# What it is
+[what-it-is]: #what-it-is
+
+
+### Proposed structure
+
+The pack commands will be reorganized in a way where they are grouped by resource type where possible. Additionally, a new
+command `config` will group subcommands that solely operate on mutating the config file.
+
+
+An overview of the (sub)commands:
+
+```
+pack
+
+  build
+  builder
+    create
+    inspect
+    suggest
+  buildpack
+    package
+    register
+  completion
+  config
+    default-builder
+      set
```

Comment:
so for this one, the command would be `pack config default-builder set <value>`?

Reply:
Yes, that's correct.

Reply:
Can we consider `pack config default-builder=<value>`?

Reply:
Is there a proposed similar solution to unsetting? Alternative to `pack config default-builder unset`.

Maybe, since it's an uncommon operation, it can simply be a flag?

```bash
# set resource
pack config default-builder=<builder-name>

# unset resource
pack config default-builder --unset

# get
pack config default-builder
```

Reply:
@jromero is this a change you're hoping to make in the RFC?

Reply:
I personally prefer `pack config default-builder set <value>` to `pack config default-builder=<builder-name>`. Using the verb allow for more natural/consistent extension if we ever want to  include other verbs. For example imagine a future where we want additional operations on set (in the other sense) values like `pack config trusted-builders add <value>`.

Reply:
If other folks _really_ don't like the verbs I will concede but even in that case I would prefer `pack config default-builder <builder-name>` to `pack config default-builder=<builder-name>`. Having an `=` in a positional argument bugs me for some reason.

Reply:
@ekcasey I think that's b/c in unix tooling you usually see it on flags and not position args. Take `ls` for instance:

```
--format=WORD
```

Reply:
Changes in this regard have been made to the RFC. Please re-review.